test/StMatic.test.ts

```diff
@@ -1326,6 +1326,124 @@ describe("Starting to test StMATIC contract", () => {
+            const checkWithdrawAndClaimAmounts = async(
+                staker: SignerWithAddress,
+                amount: BigNumber,
+                _refer: string,
+            ) => {
+                // TokenId = 1 is minted and attached to the user request.
+                const shouldClaimInMatic = (await stMATIC.convertStMaticToMatic(amount)).amountInMatic;
+                await requestWithdraw(staker, amount, _refer)
+                
+                const owned = await poLidoNFT.getOwnedTokens(staker.address);
+                const tokenToBurn = owned.at(owned.length - 1) as BigNumber;
+
+                const shouldClaimInMatic1 = await stMATIC.getMaticFromTokenId(tokenToBurn)
+                expect(shouldClaimInMatic1, "shouldClaimInMatic").eq(shouldClaimInMatic)
+
+                const withdrawalDelay = await mockStakeManager.withdrawalDelay();
+                const currentEpoch = await mockStakeManager.epoch();
+                await mockStakeManager.setEpoch(withdrawalDelay.add(currentEpoch));
+        
+                const userMaticsBeforeClaim = await mockERC20.balanceOf(staker.address);
+                await claimTokens(staker, tokenToBurn);
+                const userMaticsAfterClaim = await mockERC20.balanceOf(staker.address);
+        
+                const claimedMatics = userMaticsAfterClaim.sub(userMaticsBeforeClaim);
+                expect(claimedMatics, "claimed MATICs != requested MATICs").eq(shouldClaimInMatic);
+            }
+            
+            const totalValidatorsLength = 2;
+
+            for (let i = 0; i < totalValidatorsLength; i++) {
+                await mint(testers[i], ethers.utils.parseEther("10"));
+                await stakeOperator(testers[i]);
+                const validatorId = await mockStakeManager.getValidatorId(testers[i].address)
+                await addOperator(validatorId.toString(), testers[i].address);
+            }
+
+            const staker = user3;
+            const _refer = testers[3].address;
+
+            const userStakeAmount = toEth("100");
+            await mint(staker, userStakeAmount);
+            await submit(staker, userStakeAmount, _refer);
+
+            await stMATIC.delegate();
+
+            // The protocol is balanced now:
+            // stake: [ 50, 50 ]
+
+            // Skewing stMATIC/MATIC ratio to spot the difference later
+            await stMATIC.setProtocolFee(50)
+            const rewardToDistribute = toEth("100");
+            await mint(deployer, rewardToDistribute);
+            await mockERC20.transfer(
+                await getValidatorShareAddress(1),
+                rewardToDistribute,
+            );
+            await stMATIC.distributeRewards()
+            expect((await stMATIC.convertStMaticToMatic(100)).amountInMatic, "unexpected stMATIC ratio").eq(150);
+
+            await checkWithdrawAndClaimAmounts(staker, toEth("20"), _refer);
+            // The protocol is unbalanced now:
+            // stake: [ 30; 50 ]
```

Comment:
Isn't the stake on the first op should be 20 since we unstake 20 stMATIC, which is equal to 20 * 1.5 = 30 MATICs which should be deducted from 50 MATICs staked?

Reply:
Yes, I'll fix it!